### PR TITLE
Fix panic due to wrong intrinsic arguments (#533)

### PR DESCRIPTION
**Pull request by celinval**
During codegen of function calls, we were ignoring argument of Unit
type. This was causing a failure due to missing arguments when an
intrinsic or closure was being called with an '()' as an argument.

This change modifies how we deal with Unit types during function
definition and function call.

### Description of changes: 

Describe RMC's current behavior and how your code changes that behavior. If there are no issues this PR is resolving, explain why this change is necessary.

### Resolved issues:

Resolves #ISSUE-NUMBER

### Testing:

* How is this change tested?

* Is this a refactor change?

### Checklist
- [ ] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [ ] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: src/test/rmc/Iterator/try_fold.rs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// cbmc-flags: --unwind 20
```
Comment (danielsn) [inline src/test/rmc/Iterator/try_fold.rs:3]: Crazy that we need an unwind of 20 for this example

**Review by danielsn — APPROVED** [review]
Nice to have this out of the way!

File: src/test/rmc/Iterator/try_fold.rs (hunk 1 above)
Comment (adpaco-aws) [reply src/test/rmc/Iterator/try_fold.rs:3]: Also non-intuitive for users, which I think is a real issue. We should get this straight when we look into unwinding.

Comment (celinval) [reply src/test/rmc/Iterator/try_fold.rs:3]: That was just a random number. I can bring it down if you are worried about that.

**Pull request merged**